modules/log4cats/src/main/scala/org/legogroup/woof/log4cats/package.scala

```diff
@@ -0,0 +1,72 @@
+package org.legogroup.woof.log4cats
+
+import cats.Applicative
+import org.legogroup.woof
+import org.legogroup.woof.Logger.withLogContext
+import org.typelevel.log4cats
+
+def WoofFactory[F[_]: Applicative]: Conversion[woof.Logger[F], log4cats.LoggerFactory[F]] =
+  new Conversion[woof.Logger[F], log4cats.LoggerFactory[F]]:
+
+    def apply(logger: woof.Logger[F]): log4cats.LoggerFactory[F] =
+      new log4cats.LoggerFactory[F]:
+
+        override def fromName(name: String): F[log4cats.SelfAwareStructuredLogger[F]] =
+          Applicative[F].pure(getLoggerFromName(name))
+
+        override def getLoggerFromName(name: String): log4cats.SelfAwareStructuredLogger[F] =
+          new log4cats.SelfAwareStructuredLogger[F]:
+
+            private def logInfo(): woof.LogInfo =
+              val stacktraceElements = (new Throwable).getStackTrace()
+              val lastIndex = stacktraceElements.reverse.indexWhere(s => s.getClassName == this.getClass.getName)
+              val callingMethodIndex = stacktraceElements.size - lastIndex
+              val callingMethod      = stacktraceElements(callingMethodIndex)
+              val fileName =
+                (callingMethod.getClassName.replace('.', '/') + ".scala").split("\\/").takeRight(1).mkString
+              val lineNumber = callingMethod.getLineNumber - 1
+              woof.LogInfo(woof.EnclosingClass(name), fileName, lineNumber)
+
+            private def thrMsg(m: String, t: Throwable): String =
+              (try s"$m ${t.getMessage}"
+              catch case _ => s"$m ")
+
+            override def error(t: Throwable)(m: => String): F[Unit] = logger.error(thrMsg(m, t))(using logInfo())
+            override def warn(t: Throwable)(m: => String): F[Unit]  = logger.warn(thrMsg(m, t))(using logInfo())
+            override def info(t: Throwable)(m: => String): F[Unit]  = logger.info(thrMsg(m, t))(using logInfo())
+            override def debug(t: Throwable)(m: => String): F[Unit] = logger.debug(thrMsg(m, t))(using logInfo())
+            override def trace(t: Throwable)(m: => String): F[Unit] = logger.trace(thrMsg(m, t))(using logInfo())
+
+            override def error(m: => String): F[Unit] = logger.error(m)(using logInfo())
+            override def warn(m: => String): F[Unit]  = logger.warn(m)(using logInfo())
+            override def info(m: => String): F[Unit]  = logger.info(m)(using logInfo())
+            override def debug(m: => String): F[Unit] = logger.debug(m)(using logInfo())
+            override def trace(m: => String): F[Unit] = logger.trace(m)(using logInfo())
+
+            override def isErrorEnabled: F[Boolean] = Applicative[F].pure(true)
+            override def isWarnEnabled: F[Boolean]  = Applicative[F].pure(true)
+            override def isInfoEnabled: F[Boolean]  = Applicative[F].pure(true)
+            override def isDebugEnabled: F[Boolean] = Applicative[F].pure(true)
+            override def isTraceEnabled: F[Boolean] = Applicative[F].pure(true)
```

Comment:
Not sure if it's a problem, but log level filtering works.

Reply:
We have something similar in the slf4j-module:

https://github.com/LEGO/woof/blob/713ace4e6ba56b373f8833a1d3f1b7f56d09b991/modules/slf4j-common/src/main/scala/org/legogroup/woof/slf4j/Slf4jWoofLoggerImpl.scala#L94-L100

it should probably be reflected as a limitation in the docs